packages/at_onboarding_cli/lib/src/onboard/at_onboarding_service_impl.dart

```diff
@@ -245,12 +246,21 @@ class AtOnboardingServiceImpl implements AtOnboardingService {
           'Unable to read pkam private key from provided .atKeys path: ${atOnboardingPreference.atKeysFilePath}',
           exceptionScenario: ExceptionScenario.invalidValueProvided);
     }
-    final atChops = _createAtChops(atKeysFileDataMap);
-    await _init(atChops);
+    atChops ??= _createAtChops(atKeysFileDataMap);
+    await _init(atChops!);
     _atLookUp!.atChops = atChops;
     _atClient!.atChops = atChops;
     _atClient!.getPreferences()!.useAtChops = true;
-    _isPkamAuthenticated = await _atLookUp?.pkamAuthenticate() ?? false;
+    if (atChops.runtimeType.toString() == 'AtChopsImpl') {
+      _isPkamAuthenticated = await _atLookUp?.pkamAuthenticate() ?? false;
+    } else {
+      // Q: how we decide the signing algo type when there are more than 2 signing algo in the future?
```

Comment:
The choice could be:
1. Ask a factory to give an instance of the algorithm
2. Each signing algorithm can bind itself to a specific implementation type (could enum or string)
3. Ask the factory for an instance that matches the enum